scipy/interpolate/fitpack2.py

```diff
@@ -1977,3 +1997,16 @@ def __init__(self, u, v, r, s=0., pole_continuity=False, pole_values=None,
         self.fp = fp
         self.tck = tu[:nu], tv[:nv], c[:(nu - 4) * (nv-4)]
         self.degrees = (3, 3)
+        self.v0 = v[0]
+
+    def __call__(self, theta, phi, dtheta=0, dphi=0, grid=True):
+
+        theta = np.asarray(theta)
+        phi = np.asarray(phi)
+
+        if phi.size > 0 and (phi.min() < self.v0 or
+                             phi.max() > self.v0 + 2. * np.pi):
```

Comment:
It looks like this would be a backwards compat break - you're shifting the 2pi-wide window of values that is being accepted. It does bring the code into line with what the docs currently say for `__init__`, but that's probably only a good idea if the underlying Fortran code works like that (does it?). In that case we can consider this a bug fix. 

But we'd probably still want to check for the original [0, 2pi) range as well and either keep accepting it or emitting a deprecation warning, unless that was completely broken already?

Reply:
I'm sorry, I think I totally misunderstood the Fortran code behavior when I created this PR.
I expected the Fortran code rejects a outside point of the interpolated range, but it does not. 
The Fortran code  can evaluate any point even if the point is outside of the interpolated range:
(I did these tests with removing all input validation in `__call__`)

RectSphereBivariateSpline test results
```
>>> lats = np.array([25, 30, 35, 40, 45])
>>> lons = np.array([-90, -85, -80, -75, 70])
>>> mesh = np.meshgrid(lats, lons)
>>> data = mesh[0] * mesh[1]
>>> lat_r = np.radians(lats)
>>> lon_r = np.radians(lons)
>>> interpolator = interpolate.RectSphereBivariateSpline(lat_r, lon_r, data)
>>> query_lat = np.radians(np.array([-720, -480, -360, -180, 0, 180, 360, 480, 720]))
>>> query_lon = np.radians(np.array([-720, -480, -360, -180, 0, 180, 360, 480, 720]))
>>> interpolator(query_lat, query_lon)
array([[  1145.0000158 ,   1145.0000158 ,   1145.0000158 ,
          1145.0000158 ,   1145.0000158 ,   1145.0000158 ,
          1145.0000158 ,   1145.0000158 ,   1145.0000158 ],
       [  1145.0000158 ,   1145.0000158 ,   1145.0000158 ,
          1145.0000158 ,   1145.0000158 ,   1145.0000158 ,
          1145.0000158 ,   1145.0000158 ,   1145.0000158 ],
       [  1145.0000158 ,   1145.0000158 ,   1145.0000158 ,
          1145.0000158 ,   1145.0000158 ,   1145.0000158 ,
          1145.0000158 ,   1145.0000158 ,   1145.0000158 ],
       [  1145.0000158 ,   1145.0000158 ,   1145.0000158 ,
          1145.0000158 ,   1145.0000158 ,   1145.0000158 ,
          1145.0000158 ,   1145.0000158 ,   1145.0000158 ],
       [  1145.0000158 ,   1145.0000158 ,   1145.0000158 ,
          1145.0000158 ,   1145.0000158 ,   1145.0000158 ,
          1145.0000158 ,   1145.0000158 ,   1145.0000158 ],
       [696390.00178026, 696390.00178026, 696390.00178026,
        696390.00178026, 696390.00178026, 696390.00178026,
        696390.00178026, 696390.00178026, 696390.00178026],
       [696390.00178026, 696390.00178026, 696390.00178026,
        696390.00178026, 696390.00178026, 696390.00178026,
        696390.00178026, 696390.00178026, 696390.00178026],
       [696390.00178026, 696390.00178026, 696390.00178026,
        696390.00178026, 696390.00178026, 696390.00178026,
        696390.00178026, 696390.00178026, 696390.00178026],
       [696390.00178026, 696390.00178026, 696390.00178026,
        696390.00178026, 696390.00178026, 696390.00178026,
        696390.00178026, 696390.00178026, 696390.00178026]])
```

SmoothSphereBivariateSpline test results:
```
>>> theta = np.array([.25*pi, .25*pi, .25*pi, .5*pi, .5*pi, .5*pi, .75*pi, .75*pi, .75*pi])
>>> phi = np.array([.5 * pi, pi, 1.5 * pi, .5 * pi, pi, 1.5 * pi, .5 * pi, pi, 1.5 * pi])
>>> interpolator = interpolate.SmoothSphereBivariateSpline(theta, phi, r, s=1E10)
>>> query_lat = np.radians(np.array([-720, -480, -360, -180, 0, 180, 360, 480, 720]))
>>> query_lon = np.radians(np.array([-720, -480, -360, -180, 0, 180, 360, 480, 720]))
>>> interpolator(query_lat, query_lon)
array([[3., 3., 3., 3., 3., 3., 3., 3., 3.],
       [3., 3., 3., 3., 3., 3., 3., 3., 3.],
       [3., 3., 3., 3., 3., 3., 3., 3., 3.],
       [3., 3., 3., 3., 3., 3., 3., 3., 3.],
       [3., 3., 3., 3., 3., 3., 3., 3., 3.],
       [3., 3., 3., 3., 3., 3., 3., 3., 3.],
```

LSQSphereBivariateSpline test results.
```
>>> ntheta, nphi = 70, 90
>>> theta = np.linspace(0.5/(ntheta - 1), 1 - 0.5/(ntheta - 1), ntheta) * pi
>>> phi = np.linspace(0.5/(nphi - 1), 1 - 0.5/(nphi - 1), nphi) * 2. * pi
>>> data = np.ones((theta.shape[0], phi.shape[0]))
>>> knotst = theta[::5]
>>> knotsp = phi[::5]
>>> knotdata = data[::5, ::5]
>>> lats, lons = np.meshgrid(theta, phi)
>>> interpolator = interpolate.LSQSphereBivariateSpline(lats.ravel(), lons.ravel(), data.T.ravel(), knotst, knotsp)
>>> interpolator(query_lat, query_lon)
array([[1., 1., 1., 1., 1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1., 1., 1., 1., 1.],
       [1., 1., 1., 1., 1., 1., 1., 1., 1.]])
```

Maybe, the Fortran code uses `mod` calculation for all evaluated points.
So, I think this means the Fortran can evaluate any range of data, we don't need to check input data in `__call__' for the 3 *SphereBivariateSpline functions.
Ideally, we should remove the all input validations, but we should not do it for backward compatibility.
Do you have any good idea to cope with it?

Reply:
I'm not sure about that, it would require combing through the Fortran sources for the actual behaviour. There seem to be some constraints, for example in `parder` it says for `y` (which is `phi` in the Python code):
```
c   y     : real array of dimension (my).
c           before entry y(j) must be set to the y co-ordinate of the
c           j-th grid point along the y-axis.
c           ty(ky+1)<=y(j-1)<=y(j)<=ty(ny-ky), j=2,...,my.
```

I'd much prefer limiting this PR to just the change you tried to make to address the bug. All it needs is updating the docs and ensuring we do make the change in a backwards-compatible way.

Reply:
>I'd much prefer limiting this PR to just the change you tried to make to address the bug.
OK. I think only way to accept negative longitude and ensure backwards-compatibility is removing input validation only for longitude input. I removed the validation and fixed the test.

Reply:
This looks good to me now. Could you also update the `RectSphereBivariateSpline` docstring?